ethereum/core/src/main/java/org/hyperledger/besu/ethereum/vm/EstimateGasOperationTracer.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.vm;
+
+import org.hyperledger.besu.ethereum.core.Gas;
+import org.hyperledger.besu.ethereum.vm.ehalt.ExceptionalHaltException;
+import org.hyperledger.besu.ethereum.vm.operations.SStoreOperation;
+
+import java.util.Optional;
+
+public class EstimateGasOperationTracer implements OperationTracer {
+
+  private int maxDepth = 0;
+
+  private Gas sStoreStipendNeeded = Gas.ZERO;
+
+  @Override
+  public void traceExecution(
+      final MessageFrame frame,
+      final Optional<Gas> currentGasCost,
+      final OperationTracer.ExecuteOperation executeOperation)
+      throws ExceptionalHaltException {
+    try {
+      executeOperation.execute();
+    } finally {
+      if (frame.getCurrentOperation() instanceof SStoreOperation
+          && sStoreStipendNeeded.toLong() == 0) {
```

Comment:
Rather use `Gas.compareTo` method to check if `sStoreStipendNeeded ` is equal to zero. Conversion to long is not necessary.

Reply:
Done